### ISSUE TITLE
Panic when updating consumer config

### ISSUE THREAD

**Issue by matevzmihalic** [nats-io/nats-server#4809]
### Observed behavior

I'm updating a consumer with nats.go client (using `UpdateConsumer`) to change `FilterSubjects` field and it sometimes causes the server to crash with panic: 

```
fatal error: sync: Unlock of unlocked RWMutex
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9e4749]

goroutine 579496 [running]:
sync.fatal({0xb6bdfd?, 0x0?})
	runtime/panic.go:1061 +0x18
sync.(*RWMutex).Unlock(0xc00169c00c)
	sync/rwmutex.go:209 +0x45
panic({0xaa8e80?, 0x1041fb0?})
	runtime/panic.go:914 +0x21f
github.com/nats-io/nats-server/v2/server.(*stream).swapSigSubs(0x0, 0xc00169c000, {0xc001dcc000, 0xf0, 0x0?})
	github.com/nats-io/nats-server/v2/server/stream.go:5190 +0x49
github.com/nats-io/nats-server/v2/server.(*consumer).updateConfig(0xc00169c000, 0xc001ebcdd0)
	github.com/nats-io/nats-server/v2/server/consumer.go:1871 +0xb8d
github.com/nats-io/nats-server/v2/server.(*stream).addConsumerWithAssignment(0xc00017d180, 0xc001ebcdd0, {0x0, 0x0}, 0x0, 0x0, 0x0)
	github.com/nats-io/nats-server/v2/server/consumer.go:782 +0x1225
github.com/nats-io/nats-server/v2/server.(*stream).addConsumerWithAction(...)
	github.com/nats-io/nats-server/v2/server/consumer.go:694
github.com/nats-io/nats-server/v2/server.(*Server).jsConsumerCreateRequest(0xc000198d80, 0x0?, 0xc000ccd980, 0x3d?, {0xc002688090, 0x2a}, {0xc000620810, 0x11}, {0xc002086000, 0x3092, ...})
	github.com/nats-io/nats-server/v2/server/jetstream_api.go:3901 +0x14b4
github.com/nats-io/nats-server/v2/server.(*jetStream).apiDispatch(0xc00021c100, 0xc000218840, 0xc000ccd980, 0xc0001a3900, {0xc002688090, 0x2a}, {0xc000620810, 0x11}, {0xc002086000, 0x3092, ...})
	github.com/nats-io/nats-server/v2/server/jetstream_api.go:768 +0x25e
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc000ccd980, 0x0, 0xc000218840, 0x414705?, {0xc002688060, 0x2a, 0x30}, {0xc0006207f8, 0x11, 0x18}, ...)
	github.com/nats-io/nats-server/v2/server/client.go:3425 +0xa90
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc000ccd980, 0xc0001a3900, 0xc000f9a120, {0xc002086000, 0x3094, 0x3500}, {0x0, 0x0, 0xc00035a540?}, {0xc002688060, ...}, ...)
	github.com/nats-io/nats-server/v2/server/client.go:4477 +0xb70
github.com/nats-io/nats-server/v2/server.(*client).processServiceImport(0xc000ccd980, 0xc000240090, 0xc0001a3680, {0xc000c6c05d, 0x3043, 0x3fa3})
	github.com/nats-io/nats-server/v2/server/client.go:4262 +0x1194
github.com/nats-io/nats-server/v2/server.(*Account).addServiceImportSub.func1(0x2ad2?, 0x3000?, 0x11?, {0xc000956690?, 0xc000d86268?}, {0xc000b5ad20?, 0x1cb?}, {0xc000c6c05d, 0x3043, 0x3fa3})
	github.com/nats-io/nats-server/v2/server/accounts.go:1997 +0x2c
github.com/nats-io/nats-server/v2/server.(*client).deliverMsg(0xc000ccd980, 0x0, 0xc000219e00, 0xc001a43350?, {0xc000c6c004, 0x2a, 0x3ffc}, {0xc000c6c02f, 0x26, 0x3fd1}, ...)
	github.com/nats-io/nats-server/v2/server/client.go:3423 +0xb34
github.com/nats-io/nats-server/v2/server.(*client).processMsgResults(0xc000ccd980, 0xc0001a3680, 0xc002370330, {0xc000c6c05d, 0x3043, 0x3fa3}, {0x0, 0x0, 0x5?}, {0xc000c6c004, ...}, ...)
	github.com/nats-io/nats-server/v2/server/client.go:4477 +0xb70
github.com/nats-io/nats-server/v2/server.(*client).processInboundClientMsg(0xc000ccd980, {0xc000c6c05d, 0x3043, 0x3fa3})
	github.com/nats-io/nats-server/v2/server/client.go:3897 +0xc88
github.com/nats-io/nats-server/v2/server.(*client).processInboundMsg(0xc000ccd980?, {0xc000c6c05d?, 0x57?, 0x3ffc?})
	github.com/nats-io/nats-server/v2/server/client.go:3736 +0x37
github.com/nats-io/nats-server/v2/server.(*client).parse(0xc000ccd980, {0xc000c6c000, 0x30a0, 0x4000})
	github.com/nats-io/nats-server/v2/server/parser.go:497 +0x204f
github.com/nats-io/nats-server/v2/server.(*client).readLoop(0xc000ccd980, {0x0, 0x0, 0x0})
	github.com/nats-io/nats-server/v2/server/client.go:1371 +0x12fa
github.com/nats-io/nats-server/v2/server.(*Server).createClientEx.func1()
	github.com/nats-io/nats-server/v2/server/server.go:3217 +0x25
github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine.func1()
	github.com/nats-io/nats-server/v2/server/server.go:3700 +0x32
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine in goroutine 579507
	github.com/nats-io/nats-server/v2/server/server.go:3698 +0x145
```

This also causes the nats.go client to panic on [line 439](https://github.com/nats-io/nats.go/blob/v1.31.0/jsm.go#L439), because `info.Config` is nil. 

### Expected behavior

No panic and updated consumer config.

### Server and client version

server version: 2.10.5
nats.go version: 1.31.0

### Host environment

linux, docker, amd64

### Steps to reproduce

Update one consumer with different `FilterSubjects` concurrently.

**Comment by Jarema** [comment]
Thanks for filling the issue. Im taking a look into this one.

**Comment by Jarema** [comment]
@matevzmihalic Can you provide some additional info?

1. Do you update existing consumer, or is it create or update operation?
2. Do you update single filtered consumer into multiple filters?
3. Are the updated subjects the same, or different with each parallel update happening?

Maybe it would be easiest to answer with a client code snippet that reproduces this issue.

**Comment by matevzmihalic** [comment]
I couldn't create a small snippet of code that would reproduce this, but I've debugged this a bit more and I can tell you that stack trace I pasted above is misleading, it probably isn't related to parallel updates of a consumer and what actually happens is:

1. [stopWithFlags](https://github.com/nats-io/nats-server/blob/v2.10.5/server/consumer.go#L4948) is called which sets `o.mset` to `nil` 
2. [mset.swapSigSubs](https://github.com/nats-io/nats-server/blob/v2.10.5/server/consumer.go#L1871) then panics

I'm not sure why `stopWithFlags` is called.
